WalletWasabi/Services/WalletService.cs

```diff
@@ -253,6 +253,16 @@ private async void IndexDownloader_NewFilterAsync(object sender, FilterModel fil
 			}
 		}
 
+		public static void DropWalletCache(string walletName, string dataDir, Network network)
+		{
+			string filtersFolderPath = Path.Combine(dataDir, "BitcoinStore", network.ToString(), "IndexStore");
```

Comment:
This should not be written by hand. You should get this path somehow from the IndexStore.

Reply:
To do that I need to not make this function static and would also need to load the whole wallet service to run the command. I think this way would be cleaner.

Reply:
Yep, it seems you're right. I don't know how we could get this from IndexStore.

Reply:
You need an initialized `Global`(`InitializeNoWalletAsync`). From that you can get the `BitcoinStore` and from that the path.

Reply:
Also take this function of the `WalletService`and add it to the `BitcoinStore`'s `IndexStore` itself. Its his responsibility. (Also don't make it static.) This way when we work into the `IndexStore`, let's say we change the files we'll know that we have to adjust this, too.

Reply:
What's be also nice if you'd create an `IClearMyState` interface which would have a `ClearState` function defined. This'd be applied to `IndexStore` and `KeyManager`. To sum up:

- `WalletService.DropWalletCache` -> `IndexStore.ClearState`
- `KeyManager.AssertNetworkOrClearBlockState` -> `KeyManager.AssertNetworkOrClearBlockState`+ `KeyManager.ClearState` where `AssertNetworkOrClearBlockState` may use the `ClearState` function.